src/pad.js

```diff
@@ -0,0 +1,109 @@
+'use strict';
+
+import {Tensor} from './lib/tensor.js';
+
+/**
+ * Get mapped location from source tensor.
+ * @param {Array} location
+ * @param {Array} inputShape
+ * @param {Array} beginningPadding
+ * @param {MLPaddingMode} mode
+ * @return {Array} mapped location of source tensor
+ */
+function getMappedLocation(location, inputShape, beginningPadding, mode) {
+  const mappedLocation = location.slice();
+  const length = location.length;
```

Comment:
[naming] Consider calling this `rank` or `dimensionCount` which would make it clearer what it actually is in the `for` loop.